rand_distr/src/pert.rs

```diff
@@ -99,7 +99,7 @@ where
 
         let range = max - min;
         let mu = (min + max + shape * mode) / (shape + F::from(2.).unwrap());
-        let v = if mu == mode {
+        let v = if (mu - mode).abs() < F::epsilon() {
```

Comment:
Epsilon is not the appropriate value to use here: e.g. if min and max are very small this test will be too permissive. I'd suggest something like `2 * ε / mu` (but then it makes sense to multiply throughout by `mu`).

Reply:
Good point. I've read up on floating point comparisons (in particular [this](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/)) and force pushed some changes. I'm now using `2 * F::EPSILON * max(mu.abs(), mode.abs())`. This is something commonly suggested, i.e. scaling the machine epsilon by the maximum of the two inputs.

Reply:
This might still go wrong if one of the values is 0. The linked article talks a lot about how that's quite difficult to deal with. And actually, the popular `approx` crate treats numbers as equal if they differ less than `F::epsilon`, at least by default. (To the best of my understanding.) This is to deal with numbers close to 0. So even for ULP or relative epsilon comparison, this absolute epsilon comparison is done first. So actually, thinking about it, my initial commit might be fine for small numbers, but it's main problem was probably large numbers (where `F::epsilon` is smaller than the gap between two neighboring floats).
Uff, this is tricky :/

Reply:
I pushed a new version again. This time, the epsilon is scaled by the largest input. I think with this, everything should be fine. In particular, this deals with ["catastrophic cancellation"](https://en.wikipedia.org/wiki/Catastrophic_cancellation): where the inputs (min, max, mode, shape) are quite large, but due to subtracting them from one another, we keep the quite large rounding error even for `mu` close to 0. This is also recommended in the linked article:

> You’ll need to use an absolute epsilon, whose value might be some small multiple of FLT_EPSILON and the inputs to your calculation.